src/wp-includes/class-wp-locale-switcher.php

```diff
@@ -213,17 +209,20 @@ private function load_translations( $locale ) {
 	 *
 	 * @since 4.7.0
 	 *
-	 * @global WP_Locale $wp_locale WordPress date and time locale object.
+	 * @global WP_Locale              $wp_locale              WordPress date and time locale object.
+	 * @global WP_Textdomain_Registry $wp_textdomain_registry WordPress Textdomain Registry.
 	 *
 	 * @param string $locale The locale to change to.
 	 */
 	private function change_locale( $locale ) {
+		global $wp_locale, $wp_textdomain_registry;
+
 		// Reset translation availability information.
-		_get_path_to_translation( null, true );
+		$wp_textdomain_registry->reset();
```

Comment:
This is odd because it will remove all the stored paths for non-dotorg plugins/themes and we're at the same position as without the registry.